### PR TITLE
fix(data-table): update position of th cells to preserve border in Firefox

### PR DESCRIPTION
**Pull request by jendowns**
Partially resolves style issue 5 identified in #2330

If you look at the "selection" data table 🚨 **in Firefox** 🚨, there is something off about the first th (checkbox) and last th (`:last-of-type`) -- http://react.carbondesignsystem.com/?path=/story/datatable--with-selection

There's a 1px overflow when nothing is selected...
Screen Shot 2019-12-05 at 11 02 47 AM

And then no border when items are batch selected...
Screen Shot 2019-12-05 at 10 27 14 AM

So this PR proposes changing the position of those `th` elements from `position: relative` to the default, `position: static`. I explicitly changed it to `static`, rather than just removing the rule, because I think it's useful in this case to be explicit + include a note so anyone who comes along later understands the reasoning.

#### Changelog

**Changed**

- change position to `static` for checkbox `th` and `th:last-of-type` in heading row

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
Thanks @jendowns for jumping in! One question; Did you see there is no element with `position:absolute` _in_ those table columns? I'm asking this PR seems to change [the position calculation logic of such `position:absolute` elements](https://developer.mozilla.org/en-US/docs/Web/CSS/position) if there is.

**Comment by jendowns** [comment]
Hey @asudoh -- no, I didn't see absolutely-positioned descendants of these cells. Just text and icons 👍 (EDIT: but to clarify, I'm only testing use cases I'm aware of, namely those shown in the storybook... do you know the purpose of the `position: relative` being added on these elements originally?)

**Comment by asudoh** [comment]
I see, the history shows #2289, so if you can quickly test it's not broken I think we can go forward!

**Comment by jendowns** [comment]
Hey @asudoh, haha yes the first thing I did was check the git history 😅 The specific commits associated with these lines appear to all be related to "motion" updates, but again: there's no absolutely positioned descendants that I can see.

What I meant by my edited comment above: I'm only aware of the use case I see in the storybook. Is there another circumstance, where an absolutely positioned element maybe be added to either th cell, that I'm not aware of?

Thank you!

**Comment by asudoh** [comment]
Thanks @jendowns - To your question I don't have a good idea. If we want to dig more about this area... My question would be if you have more details on why you thought `position:relative` is the culprit?

**Comment by jendowns** [comment]
Oh I see what you mean @asudoh --

Personally anytime I see `position: relative` on a `table` child element, I start to look closer. 
Specifically I'm thinking of this detail (look under **'position'** > **relative** headings) -- https://www.w3.org/TR/CSS21/visuren.html#propdef-position

>  The effect of 'position:relative' on table-row-group, table-header-group, table-footer-group, table-row, table-column-group, table-column, table-cell, and table-caption elements is undefined. 

And if I'm not mistaken, I don't believe the spec has been updated for `position` since then.

File: packages/components/src/components/data-table/_data-table-core.scss [hunk 1]
```diff
@@ -103,7 +103,7 @@
   }
 
   .#{$prefix}--data-table th:last-of-type {
-    position: relative;
+    position: static; // Required to preserve border.
```
Comment (joshblack) [inline packages/components/src/components/data-table/_data-table-core.scss]: Super unimportant style-nit lol:

```suggestion
    // Required to preserve border
    position: static;
```

**Comment by jendowns** [comment]
Thank you @joshblack I just updated the formatting of my comments!
Also was curious if you thought the comments were sufficient? Based on the discussion here, I wonder if I shouldn't include more info on the spec (see: https://github.com/carbon-design-system/carbon/pull/4822#issuecomment-562356444)?

Maybe something that references less of one "effect" (i.e., "Required to preserve borders") and more of the cause? Like this:

```
// Do not use `position: relative`, as its behavior is undefined: https://www.w3.org/TR/CSS21/visuren.html#propdef-position
```

Too much?

**Comment by joshblack** [comment]
@jendowns always down to include more info over less, to be honest, but this is 100% my personal preference haha. Especially if it's something crazy like this.

**Comment by joshblack** [comment]
One quick note to reviewers, this is observable in Firefox 👍

**Review by joshblack — APPROVED** [review]
Looks great! Would love to see your additional comments too, @jendowns if you wanted to add them

**Comment by jendowns** [comment]
Thank you @joshblack! Just updated the comment to include those details 👍

**Review by dakahn — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
💯 explanation - Thanks @jendowns!

**Review by aledavila — APPROVED** [review]
Thanks Jen. Good PR

**Pull request merged**